sdks/python/apache_beam/io/filesystem.py

```diff
@@ -219,28 +219,33 @@ def _initialize_compressor(self):
 
   def readable(self):
     # type: () -> bool
-    mode = self._file.mode
-    return 'r' in mode or 'a' in mode
+    try:
+      return 'r' in self._file.mode or 'a' in self._file.mode
+    except AttributeError:
+      return self._file.readable is True
 
   def writeable(self):
     # type: () -> bool
-    mode = self._file.mode
-    return 'w' in mode or 'a' in mode
+    try:
+      return 'w' in self._file.mode or 'a' in self._file.mode
+    except AttributeError:
+      return self._file.writable is True
 
   def write(self, data):
     # type: (bytes) -> None
 
     """Write data to file."""
     if not self._compressor:
-      raise ValueError('compressor not initialized')
+      self._initialize_compressor()
```

Comment:
`self._initialize_compressor` is done in constructor if the file is writable. If hit this code path there is something wrong with the config of the writeable() (see above).

In general, it should not be necessary to change the logics of filesystem.py here

Reply:
As discussed, I'll be restoring this file and implementing a wrapper class for BlobReader and BlobWriter in gcsio.py